code/game/objects/items/robot/robot_upgrades.dm

```diff
@@ -947,3 +947,133 @@
 		var/obj/item/borg_snack_dispenser/medical/lollipopshooter = new(R.module)
 		R.module.basic_modules += lollipopshooter
 		R.module.add_module(lollipopshooter, FALSE, TRUE)
+
+
+/obj/item/borg/upgrade/engi_advancedtools
+	name = "engineering cyborg advanced tool kit"
+	desc = "An upgrade for engineering cyborgs which replaces their basic tools with an advanced verison of them."
+	icon_state = "cyborg_upgrade3"
+	require_module = TRUE
+	module_type = /obj/item/robot_module/engineering
+	module_flags = BORG_MODULE_ENGINEERING
+
+/obj/item/borg/upgrade/engi_advancedtools/action(mob/living/silicon/robot/R, user = usr)
+	. = ..()
+	if(!.)
+		return FALSE
+
+	for(var/obj/item/screwdriver/cyborg/SC in R.module.modules) 
+		R.module.remove_module(SC, TRUE)
+
+	for(var/obj/item/wrench/cyborg/W in R.module.modules) 
+		R.module.remove_module(W, TRUE)
+
+	for(var/obj/item/crowbar/cyborg/CB in R.module.modules) 
+		R.module.remove_module(CB, TRUE)
+
+	for(var/obj/item/wirecutters/cyborg/WC in R.module.modules) 
+		R.module.remove_module(WC, TRUE)
+
+	for(var/obj/item/multitool/cyborg/MT in R.module.modules) 
+		R.module.remove_module(MT, TRUE)
+	
+	for(var/obj/item/analyzer/AL in R.module.modules) 
+		R.module.remove_module(AL, TRUE)
+
+	var/obj/item/jawsoflife/JL = locate() in R.module.modules
+	var/obj/item/handdrill/HD = locate() in R.module.modules
+	var/obj/item/multitool/tricorder/TC = locate() in R.module.modules
+	if(JL || HD || TC)
+		to_chat(user, span_warning("This cyborg is already equipped with an advanced engineering tool kit."))
+		return FALSE
+
+	JL = new(R.module)
+	JL.toolspeed = 0.5 // Same toolspeed as the cyborg crowbar/cutters.
```

Comment:
This is not what I meant. Make them cyborg subtypes please.

Reply:
Done.